fog/test-client/src/test_client.rs

```diff
@@ -1545,8 +1545,14 @@ impl HealthTracker {
         // * there is a failure
         // * the failure happened at least healing_time ago then set ourselves healthy
         //   again
+        // The combination of have_failure + last_failure can be thought of as an
+        // `Option<u64>`, but because there's no Atomic version of
+        // `Option<u64>`, we need to have two atomics. If we had an
+        // AtomicOptionU64 type then we would check for Some(last_failure), and we would
+        // never set last_failure to None, which is also why we don't set have_failure
+        // to false.
         if self.have_failure.load(Ordering::SeqCst)
-            && self.last_failure.load(Ordering::SeqCst) + self.healing_time <= counter
+            && self.last_failure.load(Ordering::SeqCst) + self.healing_time < counter
         {
             counters::LAST_POLLING_SUCCESSFUL.set(1);
```

Comment:
suggest getting rid of the `have_failure`, it's adding more logic than necessary
If it was clarifying the logic, then I could see keeping it, but the addition of this comment and the fact it never gets set back to false seems to me this logic would be simpler only having `last_failure`

with `healing_time` of 1
1. System is initialized as healthy
  a. LAST_POLLING_SUCCESSFUL is set to 1
  b. `last_failure` is 0
  c. `count` is 0
2. If `announce_failure` get's called on first run
  a. It will set `LAST_POLLING_SUCCESSFUL` to 0
  b. `last_failure` will stay at 0, then the `count` will get incremented to 1
  c. set_counter() will be called with `count` of 1. `LAST_POLLING_SUCCESSFUL` will remain unchanged.
3. `count` increments to 2 and set_counter() is called
  a. `LAST_POLLING_SUCCESSFUL` will be set back to 1
  
  
If `announce_failure()` doesn't get called on first run, the `LAST_POLLING_SUCCESSFUL` will stay in the initial state of 1

```suggestion
        if self.last_failure.load(Ordering::SeqCst) + self.healing_time < counter {
            counters::LAST_POLLING_SUCCESSFUL.set(1);
```

Reply:
That's fine by me.

Reply:
It does mean that `last_failure` is ambiguous when it is set to zero - it's not possible to tell just by looking at it whether a failure occurred at the first transaction or not, but since it's not exposed outside of `HealthTracker` then I suppose that's acceptable.

Reply:
you could also just do `Mutex<Option<u64>>`, there's no particularly good reason to use atomics here